### ISSUE TITLE
Is there any way to set the region in Java Code?

### ISSUE THREAD

**Issue by EvgenyKislyy** [aws/amazon-s3-encryption-client-java#183]
Hello, is there any way to set the region in Java code?
We are using primary and secondary regions, but as I understand - now it's possible to set only one region via 
`export AWS_REGION="us-west-2"`
So I can't understand how we can use two clients: one for the primary region, and the second for the secondary

**Comment by arikogan** [comment]
Hi. Thanks for the above question and answer.
A question about this code:
```
Keyring kmsKeyring = KmsKeyring.builder()
        .kmsClient(KmsClient.builder().region(US_EAST_1).build())
        .wrappingKeyId(KMS_KEY_ID)
        .build();
```
When calling `KmsKeyring.builder()` a default KmsClient is [instantiated](https://github.com/aws/amazon-s3-encryption-client-java/blob/main/src/main/java/software/amazon/encryption/s3/materials/KmsKeyring.java#L233) and only after it is replaced by the provided KmsClient with the region we want (`.kmsClient(anotherClient)`).
The thing is that the `DefaultKmsClient` requires a region to be instantiated, but the region we want is set programmatically afterwards. The workaround is to do something like `System.setProperty("aws.region", Region.US_EAST_1.id())` to avoid the `DefaultAwsRegionProviderChain` failing.
I'm thinking of the possibility of delaying the creation of the `DefaultKmsClient` in the `KmsKeyring` subjected to if another one was not set. Is there any reason not to do this?
Thanks!

**Comment by JoergHeinicke5005** [comment]
@imabhichow Is it possible to get the fix released? I'm also running into the issue that creation of S3AsyncEncryptionClient fails with region being not set while I intend to pass a properly wrapped client to its builder instead.

**Comment by imabhichow** [comment]
Hello @EvgenyKislyy,

Thank you for the report.

We have published a fix in new version [3.1.2](https://github.com/aws/amazon-s3-encryption-client-java/releases/tag/v3.1.2).

Thank you!

**Comment by avishayh** [comment]
Thanks
